### PR TITLE
Allow creation of partners from *Create and Edit*.

### LINKED ISSUE #78: [8.0] partner_firstname and "create and edit" in M2O widget to res.partner
Here are the steps to reproduce the bug on odoo v8 with an up-to-date branch of OCA/partner-contact/8.0:

1) Create a new sale order
In the "Customer" field (M2O to res.partner), type "TOTOTEST1" (it should not be an existing partner) and then, in the down-drop list that appear, click on "Create and Edit..."
2) You get a pop-up that displays the form view of res.partner with:
Name = TOTOTEST1
Firstname = False
Lastname = False
3) in the partner pop-up, enter:
Lastname = de Lattre
Fistname =  Alexis
Street = 42 rue de la paix
and click on "Save" in the partner pop-up and save in the quotation form.
4) Go to the "Customers" menu. You will find a partner with:
Name = TOTOTEST1
Lastname = TOTOTEST1
Firstname = False
Street = 42 rue de la paix

Conclusion : when the module partner_firstname is installed, we have a bug with "Create and Edit..." in the M2O widget that point to res.partner.

### PR DESCRIPTION
**Pull request by yajo**
This fixes #78 and adds new tests for it.

### CODE REVIEW

File: partner_firstname/tests/test_create.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        self.context["default_name"] = u"BÄD2"
+
+
+class CompanyCase(PersonCase):
+    """Test ``res.partner`` when it is a company."""
+    context = {"default_is_company": True}
+
+    def setUp(self):
+        return super(CompanyCase, self).setUp()
+        self.values.update(lastname=self.values["name"], firstname=False)
```
Comment (moylop260) [inline partner_firstname/tests/test_create.py]: sentence after return

**Comment by moylop260** [comment]
after fix travis builds :+1: 

``` bash
partner_firstname/tests/base.py:74:80: E501 line too long (81 > 79 characters)
partner_firstname/tests/test_create.py:66: [W0101(unreachable), CompanyCase.setUp] Unreachable code
```

**Comment by yajo** [comment]
Thanks @moylop260, it should be ready now.

**Comment by rafaelbn** [comment]
Tested and fixed :+1:

**Comment by pedrobaeza** [comment]
:+1:

**Comment by alexis-via** [comment]
@Yajo Thank you very much for this bug fix ! You are THE man !

**Pull request merged**